api/renter.go

```diff
@@ -275,17 +331,57 @@ func (api *API) renterUploadHandler(w http.ResponseWriter, req *http.Request, ps
 		return
 	}
 
+	// Check whether the erasure coding parameters have been supplied.
+	var ec modules.ErasureCoder
+	if req.FormValue("datapieces") != "" || req.FormValue("paritypieces") != "" {
+		// Check that both values have been supplied.
+		if req.FormValue("datapieces") == "" || req.FormValue("paritypieces") == "" {
+			WriteError(w, Error{"Must provide both the datapieces paramaeter and the paritypieces parameter if specifying erasure coding parameters"}, http.StatusBadRequest)
+			return
+		}
+
+		// Parse the erasure coding parameters.
+		var dataPieces, parityPieces int
+		_, err := fmt.Sscan(req.FormValue("datapieces"), &dataPieces)
+		if err != nil {
+			WriteError(w, Error{"Unable to read parameter 'datapieces': " + err.Error()}, http.StatusBadRequest)
+			return
+		}
+		_, err = fmt.Sscan(req.FormValue("paritypieces"), &parityPieces)
+		if err != nil {
+			WriteError(w, Error{"Unable to read parameter 'paritypieces': " + err.Error()}, http.StatusBadRequest)
+			return
+		}
+
+		// Verify that sane values for parityPieces and redundancy are being
+		// supplied.
+		if parityPieces < requiredParityPieces {
+			WriteError(w, Error{fmt.Sprintf("A minimum of %v parity pieces is required, but %v parity pieces requested.", parityPieces, requiredParityPieces)}, http.StatusBadRequest)
+		}
+		redundancy := float64(dataPieces+parityPieces) / float64(dataPieces)
+		if float64(dataPieces+parityPieces)/float64(dataPieces) < requiredRedundancy {
+			WriteError(w, Error{fmt.Sprintf("A redundancy of %.2f is required, but redundancy of %.2f supplied", redundancy, requiredRedundancy)}, http.StatusBadRequest)
+			return
+		}
+
+		// Create the erasure coder.
+		ec, err = renter.NewRSCode(dataPieces, parityPieces)
+		if err != nil {
+			WriteError(w, Error{"Unable to encode file using the provided parameters: " + err.Error()}, http.StatusBadRequest)
+			return
+		}
+	}
```

Comment:
Should be noted that the testing here is not really complete. It's probably in our interest not to merge this until coverage is at 100%.